src/common/src/array/primitive_array.rs

```diff
@@ -48,15 +49,15 @@ where
     fn try_into_array_ref(arr: &ArrayImpl) -> Option<&PrimitiveArray<Self>>;
     /// Returns array type of the primitive array
     fn array_type() -> ArrayType;
-    /// Creates an `ArrayBuilder` for this primitive type
-    fn create_array_builder(capacity: usize) -> ArrayBuilderImpl;
 
     // item methods
     fn to_protobuf<T: Write>(self, output: &mut T) -> ArrayResult<usize>;
 }
 
 macro_rules! impl_array_methods {
     ($scalar_type:ty, $array_type_pb:ident, $array_impl_variant:ident) => {
+        const DATA_TYPE: DataType = DataType::$array_impl_variant;
```

Comment:
FYI there is one exception that `array_impl_variant`is different from `DataType` variant: `Utf8` vs `Varchar`. But it is not primitive and we are good here.